helper.py

```diff
@@ -1,41 +1,115 @@
+import aionotify
+from websockets.server import serve
+import websockets
+import time
+import base64
+import os
+from io import BytesIO
 from PIL import Image
 
 
-static_path = 'shepherd/static/'
-input_file = static_path + "full_image.jpg"
-output_file = static_path + "image.jpg"
+CONNECTIONS = set()
+
+#The following sets up the asynchronous waiting for file change
+watcher = aionotify.Watcher()
+
+#TODO - Change the input_file to the location of the image the camera takes!
+static_path = '/home/pi/shepherd/shepherd/static/'
+input_file = static_path + "image.jpg"
+output_file = static_path + "resized_image.jpg"
 stat_file = static_path + "imgtime.txt"
 mod_time = 0
 out_width = 640.0
 out_height = 480.0
 
-while True:
-    time.sleep(0.1)
-    if os.path.exists(input_file):
-        stat_obj = os.stat(input_file)
-        if mod_time != stat_obj.st_mtime:
-            try:
-                img = Image.open(input_file)
-                width_i, height_i = img.size
-                width = width_i * 1.0
-                height = height_i * 1.0
-                scale_x = out_width / width
-                scale_y = out_height / height
-                if scale_x > scale_y:
-                    img = img.resize((int(width*scale_y), int(height*scale_y)), Image.ANTIALIAS)
-                else:
-                    img = img.resize((int(width*scale_x), int(height*scale_x)), Image.ANTIALIAS)
-
-                mod_time = stat_obj.st_mtime
-                img.save(output_file,"JPEG")
-                with open(stat_file, "w") as myfile:
-                    myfile.write(str(mod_time))
-            except IOError:
-                pass
+watcher.watch(alias='image', path=input_file, flags=aionotify.Flags.MODIFY)#sets up watcher
+
+def im_2_b64(image):
+    buff = BytesIO()
+    image.save(buff, format="JPEG")
+    img_str = base64.b64encode(buff.getvalue())
+    return img_str
+
+async def wait_for_file_change():
+    loop = asyncio.get_event_loop()
+
+    bypass = False#so first image is not ignored.
+    while not os.path.exists(input_file):
+        time.sleep(0.5)#twiddle thumbs :)
+        if not bypass:
+            bypass = True
+    await watcher.setup(loop)
+    print("File change watcher is running.")
+    
+    while True:#for all events
+        if not bypass:
+            event = await watcher.get_event()#blocks until file changed
+        else:
+            bypass = False#reset bypass
+        failing = True
+        c = 0
+        while failing:
+            time.sleep(0.1)#give it time to write the file.
+            if c < 3:
+                try:#this runs until the bot has finished writing the image
+                    img = Image.open(input_file)
+                    img.load()
+                    print("Opened successfully")
+                    failing = False
+                except:
+                    print("Error opening file: attempt #"+str(c))
+            else:
+                failing = False
+            c += 1
```

Comment:
Thinking about this maybe it would be better off as a for loop which we break out of, with the continue in an else